### PR TITLE
Disable synchronous_commit by default for writer connections

### PR DESCRIPTION
**Pull request by jgpruitt**
## Description

Disabling synchronous_commit for the database connections in the writer pool will improve performance. The tradeoff is that some amount of writes may be lost in the event of a crash. This is likely an "okay" tradeoff for monitoring data.
By default, we now disable synchronous_commit on connections in the writer pool at the session level. This does not impact other connections/sessions. A new config option -- `db.connections.writer-pool.synchronous-commit` -- has been added to override this default.

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [x] CHANGELOG entry for user-facing changes
- [x] Updated the relevant documentation

### CODE REVIEW

**Review by antekresic — COMMENTED** [review]
Couple of suggestions.

File: pkg/tests/end_to_end_tests/sync_commit_test.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		return setting
+	}
+
+	withDB(t, "writer_sync_commit", func(db *pgxpool.Pool, t testing.TB) {
+		var lockerCalled bool // used to ensure that the schema locker function is still called
+
+		// create a writer pool with synchronous_commit turned off
+		writerPool := createWriterPool(db.Config().ConnString(), &lockerCalled, false)
+		if getSynchronousCommit(writerPool) != "off" {
+			t.Failed()
```
[the remark this answers is not in the document]
Comment (jgpruitt) [reply pkg/tests/end_to_end_tests/sync_commit_test.go:53]: yep. i'll change that.

File: pkg/pgclient/client.go [hunk 2]
```diff
@@ -92,7 +92,7 @@ func NewClient(r prometheus.Registerer, cfg *Config, mt tenancy.Authorizer, sche
 		if err != nil {
 			return nil, fmt.Errorf("get writer pg-config: %w", err)
 		}
-		writerPgConfig.AfterConnect = schemaLocker
+		SetWriterPoolAfterConnect(writerPgConfig, schemaLocker, cfg.WriterSynchronousCommit)
```
[the remark this answers is not in the document]
Comment (antekresic) [reply pkg/pgclient/client.go:95]: Yeah, now I see that `NewClient` is getting pretty complex with zero tests. 

So for this PR, I'm fine doing it as you did it and adding a TODO to add some testing to `NewClient`.

File: pkg/tests/end_to_end_tests/sync_commit_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jgpruitt) [reply pkg/tests/end_to_end_tests/sync_commit_test.go]: done

**Review by antekresic — APPROVED** [review]

**Review by alejandrodnm — APPROVED** [review]

**Pull request merged**